### ISSUE TITLE
(Slide) curtains not showing correct status

### ISSUE THREAD

**Issue by meijerpeter** [swttt/com.swttt.homekit#202]
Hi,

I have added the Siide curtains to the Homey and enabled them via Homekit in my Home app. The commands are getting through correctly, but the status of the curtains in the Home app is always "closed". In the Homey app I have the correct status.

For example, if I tell Home to set the curtains to 50% they are at 50%, but in the Home app the status is "closed". In Homey the 50% is shown.

All apps are updated to their latest version (Homey 7.1.3, Homeykit v3.0.16). 

Let me know if you need any diagnostic data.

Thanks for your amazing work!!

**Comment by robertklep** [comment]
The app probably uses multiple capabilities that are conflicting (at least in how Homeykit interprets them).

You can find the device and the current state of its capabilities here: https://tools.developer.homey.app/tools/devices

I'm interested in the values of the capabilities when you get an incorrect status.

**Comment by meijerpeter** [comment]
Completely closed:
<img width="939" alt="Screenshot 2021-09-25 at 16 10 46" src="https://user-images.githubusercontent.com/7052335/134774550-04e70ca1-c6f9-4389-b90e-fdb18e735a7b.png">

Completely open:
<img width="938" alt="Screenshot 2021-09-25 at 16 12 16" src="https://user-images.githubusercontent.com/7052335/134774551-de15bd01-bb37-43a6-97c3-0ce71a838b18.png">

Set to 56%:
<img width="940" alt="Screenshot 2021-09-25 at 16 13 10" src="https://user-images.githubusercontent.com/7052335/134774554-c4445421-a23c-47b3-b57f-307bcce68501.png">

**Comment by robertklep** [comment]
I'm not sure how Home decides if a curtain is closed or not. From what I can see, there's nothing in Homeykit that sets that state.

**Comment by meijerpeter** [comment]
Update: the closed curtain is displaying 8% open. The value in Homey is 0.08. The other is closed and has value o.97. I think Slide has their numbers inverted…. Hassio had the same problem, see their option at the bottom: https://www.home-assistant.io/integrations/slide/

I will ask the developers of the Slide plugin if they can change this behaviour, or is this something you can change?

**Comment by robertklep** [comment]
Both in HomeKit and in Homey, 0 should mean "fully closed" and 100 (or "1" in Homey) should mean "fully open". That's what Homeykit assumes as well.

What might be an issue though is that the Homey values don't to all the way to 0 and 1. I can imagine that Home might consider any value _not_ 1 to be "closed" (or at least _partially_ closed).

**Comment by meijerpeter** [comment]
I agree, I don’t see whole numbers. You might be helping out to round the open and close states when the value is “almost” 0 or 1, e.g. 0.97 can be rounded to open?

**Comment by robertklep** [comment]
Rounding would work in your situation, but I have no idea if that would break things for other people. So ideally, the issue should be fixed in the app.

**Comment by meijerpeter** [comment]
Agreed!
